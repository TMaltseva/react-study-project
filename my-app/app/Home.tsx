'use client';

import React, { useState } from 'react';
import { useRouter, useSearchParams } from 'next/navigation';
import { useFetchPeopleQuery } from '../services/api';
import SearchBar from '../components/SearchBar';
import SearchResults from '../components/SearchResults';
import ErrorButton from '../components/ErrorButton';
import Pagination from '../components/Pagination';
import DetailsWrapper from '../components/DetailsWrapper';
import ThemeToggleButton from '../components/ThemeToggleButton';
import Flyout from '../components/Flyout';

const Home: React.FC = () => {
  const router = useRouter();
  const searchParams = useSearchParams();
  const searchTerm = searchParams.get('search') || '';
  const page = parseInt(searchParams.get('page') || '1', 10);
  const { data, isLoading, isFetching } = useFetchPeopleQuery({ searchTerm, page });
  const [selectedId, setSelectedId] = useState<string | null>(null);

  const handleSearch = (term: string, page: number = 1) => {
    const params = new URLSearchParams({ page: page.toString(), search: term });
    router.push(`/?${params.toString()}`);
  };

  const handlePageChange = (page: number) => {
    const params = new URLSearchParams({ page: page.toString(), search: searchTerm });
    router.push(`/?${params.toString()}`);
  };

  const handleItemClick = (id: string) => {
    setSelectedId(id);
  };

  const handleCloseDetails = () => {
    setSelectedId(null);
  };

  return (
    <main className="sections-wrapper">
      <div className="top-section">
        <SearchBar onSearch={(term) => handleSearch(term, 1)} />
        <ErrorButton />
        <ThemeToggleButton />
        <Flyout />
      </div>
      <div className="bottom-section">
        <div className="left-section">
          {isLoading || isFetching ? (
            <p>Loading...</p>
          ) : (
            <SearchResults results={data?.results || []} onItemClick={handleItemClick} />
          )}
        </div>
        <div className="right-section">
          {selectedId && <DetailsWrapper id={selectedId} onClose={handleCloseDetails} />}
        </div>
      </div>
      {data?.results?.length ? (
        <Pagination currentPage={page} totalPages={data.totalPages} onPageChange={handlePageChange} />
      ) : null}
    </main>
  );
};

export default Home;
