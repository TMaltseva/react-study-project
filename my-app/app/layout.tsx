import React from 'react';
import './styles/globals.css';

export default function Layout({ children }: { children: React.ReactNode }) {
  return (
    <html>
      <head>
        <link rel="icon" href="/app/favicon.ico" />
      </head>
      <body>{children}</body>
    </html>
  );
}
