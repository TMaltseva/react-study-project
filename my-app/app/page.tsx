import ClientApp from './components/ClientApp';

export default function Page() {
  return <ClientApp />;
}
