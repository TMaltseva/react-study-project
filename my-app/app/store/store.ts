import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import { api } from '../services/api';
import selectedItemsReducer from '../services/selectedItemsSlice';
import { createWrapper } from 'next-redux-wrapper';

export const makeStore = () =>
  configureStore({
    reducer: {
      [api.reducerPath]: api.reducer,
      selectedItems: selectedItemsReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware),
  });

setupListeners(makeStore().dispatch);

export type RootState = ReturnType<ReturnType<typeof makeStore>['getState']>;
export type AppDispatch = ReturnType<typeof makeStore>['dispatch'];

export const wrapper = createWrapper(makeStore);

export const store = makeStore();
