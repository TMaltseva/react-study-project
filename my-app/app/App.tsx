'use client';

import React from 'react';
import { wrapper } from '../store/store';
import { Provider } from 'react-redux';
import Home from './Home';

const App: React.FC = () => {
  const { store } = wrapper.useWrappedStore({});
  return (
    <Provider store={store}>
      <Home />
    </Provider>
  );
};

export default App;
