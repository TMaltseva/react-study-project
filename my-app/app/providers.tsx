'use client';

import React, { ReactNode } from 'react';
import { Provider as ReduxProvider } from 'react-redux';
import { store } from './store/store';
import ThemeProvider from './components/ThemeProvider';

export function Providers({ children }: { children: ReactNode }) {
  return (
    <ReduxProvider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </ReduxProvider>
  );
}
