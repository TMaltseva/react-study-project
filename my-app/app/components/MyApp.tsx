import React from 'react';
import { Provider } from 'react-redux';
import { AppProps } from 'next/app';
import { wrapper } from '../store/store';
import ThemeProvider from './ThemeProvider';

const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
  const { store, props } = wrapper.useWrappedStore(pageProps);

  return (
    <Provider store={store}>
      <ThemeProvider>
        <Component {...props.pageProps} />
      </ThemeProvider>
    </Provider>
  );
};

export default MyApp;
