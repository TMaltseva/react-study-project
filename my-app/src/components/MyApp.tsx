import React from 'react';
import { Provider } from 'react-redux';
import { AppProps } from 'next/app';
import { wrapper } from '../store/store';
import ThemeProvider from './ThemeProvider';
import ErrorBoundary from './ErrorBoundary';
import '../styles/globals.css';

const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
  const { store, props } = wrapper.useWrappedStore(pageProps);

  return (
    <ErrorBoundary>
      <Provider store={store}>
        <ThemeProvider>
          <Component {...props.pageProps} />
        </ThemeProvider>
      </Provider>
    </ErrorBoundary>
  );
};

export default MyApp;
