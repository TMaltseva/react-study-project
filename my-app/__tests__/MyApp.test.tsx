import { render } from '@testing-library/react';
import { AppProps } from 'next/app';
import { wrapper } from '../src/store/store';
import MyApp from '../src/components/MyApp';
import '@testing-library/jest-dom';
import { Router } from 'next/router';

jest.mock('../src/store/store', () => ({
  wrapper: {
    useWrappedStore: jest.fn().mockReturnValue({
      store: {},
      props: { pageProps: {} },
    }),
  },
}));

const mockComponent = () => <div>Mock Component</div>;

describe('MyApp', () => {
  it('renders correctly', () => {
    const props: AppProps = {
      Component: mockComponent,
      pageProps: {},
      router: {
        basePath: '',
        pathname: '/',
        route: '/',
        asPath: '/',
        query: {},
        push: jest.fn(),
        replace: jest.fn(),
        reload: jest.fn(),
        back: jest.fn(),
        prefetch: jest.fn().mockResolvedValue(undefined),
        beforePopState: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
          emit: jest.fn(),
        },
        isFallback: false,
        isReady: true,
        isLocaleDomain: false,
        isPreview: false,
        components: {},
        sdc: {},
        sbc: {},
        sub: jest.fn(),
      } as unknown as Router,
    };

    const { asFragment } = render(<MyApp {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});