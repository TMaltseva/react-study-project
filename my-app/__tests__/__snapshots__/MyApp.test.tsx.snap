// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApp renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="error-message"
    >
      Something went wrong
    </h2>
    <button>
      Go Search
    </button>
  </div>
</DocumentFragment>
`;
